rust/foxglove/src/websocket.rs

```diff
@@ -972,6 +983,50 @@ impl ConnectedClient {
         let message = Message::binary(buf);
         self.send_control_msg(message);
     }
+
+    /// Advertises a channel to the client.
+    fn advertise_channel(&self, channel: &Arc<Channel>) {
```

Comment:
Do you think it's time for ConnectedClient to get its own module/file? This doesn't bother me with IDE support, but maybe would make the context more obvious when reading.

Reply:
+1 I've considering this for a while, the file is getting unwieldy for me at this length